debian/marquee/nginx/sites/books.spec.whatwg.org.conf

```diff
@@ -0,0 +1,13 @@
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
```

Comment:
Can we move these to whatwg.conf?

Reply:
Or http.conf? What's the difference between http and whatwg?

Reply:
I think http.conf is for not-HTTPS.

Reply:
listen 443 ssl http2;
    listen [::]:443 ssl http2;

 > Can we move these to whatwg.conf?

No we can’t. Each server block needs to have its own `listen` directive; otherwise the server gets defaulted to port 80:

https://nginx.org/en/docs/http/ngx_http_core_module.html#listen
> If the directive is not present then either *:80 is used if nginx runs with the superuser privileges, or *:8000 otherwise.

https://www.digitalocean.com/community/tutorials/understanding-nginx-server-and-location-block-selection-algorithms#parsing-the-quot-listen-quot-directive-to-find-possible-matches

>  By default, any server block that does not include a listen directive is given the listen parameters of `0.0.0.0:80` (or `0.0.0.0:8080` if Nginx is being run by a normal, non-root user).

And there’s no config option to override that; it’s hardcoded into the nginx sources:

https://github.com/nginx/nginx/blob/3900d1cb3cf39963d909604923f7e7af6ecf99fd/src/http/ngx_http_core_module.c#L2775-L2784

Reply:
> Or http.conf? What's the difference between http and whatwg?
>
> I think http.conf is for not-HTTPS.

Yeah, the only purpose of  `http.conf` is just to redirect non-HTTPS to HTTPS.

But as far as why both `http.conf` and `whatwg.conf` are needed: because `http.conf` gets put into the `/etc/nginx/conf.d` directory, and anything that’s put there gets included once in the main top-level nginx `http {…}` block. But the stuff in `whatwg.conf` needs to get included into multiple places, in the `server {…}` blocks for each virtual server (domain).

Reply:
> Each server block needs to have its own listen directive

Huh, and including such a directive via `include` doesn't mean "having" it?

Reply:
> > Each server block needs to have its own listen directive
>
> Huh, and including such a directive via `include` doesn't mean "having" it?

Ah, I hadn’t thought about just putting it in `whatwg.conf`. But I moved it there and it works find that way, so I’ll commit that change—thanks

Reply:
> Did that work out, can you push those changes?

It did work, and I’ve now pushed the changes I‘ve made so far. I will wrap up the rest and push them later today.